### PR TITLE
feat: Yolov5 Object detector UDF

### PR DESCRIPTION
**Pull request by Aryan-Rajoria**
Adding Yolov5 Object Detector UDF.
About 100x faster than `fasterRCNNObjectDetector`, when used with EVA.

### CODE REVIEW

File: eva/udfs/yolo_object_detector.py [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+        Returns:
+            tuple containing predicted_classes (List[List[str]]),
+            predicted_boxes (List[List[BoundingBox]]),
+            predicted_scores (List[List[float]])
+
+        """
+        # img = frames[0].transpose()
+        
+        img_b = frames[0][0]
```
Comment (gaurav274) [inline eva/udfs/yolo_object_detector.py:177]: Are you assuming input contains only one frame?

File: eva/udfs/yolo_object_detector.py [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+        img_g = frames[0][1]
+        img_r = frames[0][2]
+        img_rgb = torch.stack((img_r, img_g, img_b), dim=2)
+
+        img_rgb = img_rgb.cpu().detach().numpy()
+        img_rgb = img_rgb*255
+        print(type(img_rgb))
+
+        predictions = self.model(img_rgb)
+        predictions.print()
```
Comment (gaurav274) [inline eva/udfs/yolo_object_detector.py:187]: Please remove the print statements.

File: setup.py [hunk 3]
```diff
@@ -97,7 +96,14 @@ def read(path, encoding="utf-8"):
 ### NEEDED FOR A BATTERIES-LOADED EXPERIENCE
 udf_libs = [
     "facenet-pytorch>=2.5.2",
-    "easyocr>=1.5.0"
+    "easyocr>=1.5.0",
+    "ipython",
```
Comment (gaurav274) [inline setup.py:100]: Are these requirements of `yolo`? Can we get remove some of these?

**Review by gaurav274 — CHANGES_REQUESTED** [review]

**Comment by gaurav274** [comment]
Also, please add test cases for this UDF. You can add testcases similar to FasterRCNN. Look at examples in `integration_tests`

File: eva/udfs/yolo_object_detector.py (hunk 2 above)
Comment (Aryan-Rajoria) [reply eva/udfs/yolo_object_detector.py:187]: Thank you, for pointing that out, will remove that.

File: setup.py (hunk 3 above)
Comment (Aryan-Rajoria) [reply setup.py:100]: I have removed a few dependencies

**Comment by jarulraj** [comment]
Can we get rid of FasterRCNN at the same time?

File: setup.py [hunk 4]
```diff
@@ -97,7 +96,13 @@ def read(path, encoding="utf-8"):
...
+    # "psutil",
```
Comment (jarulraj) [inline setup.py:101]: I noticed these Tensorflow dependencies. Are you able to add a Tensorflow-based UDF?

File: test/udfs/test_yolo_object_detector.py [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+import sys
+import unittest
+
+import cv2
+import mock
+import pandas as pd
+
+from eva.models.storage.batch import Batch
+
+NUM_FRAMES = 10
```
Comment (jarulraj) [inline test/udfs/test_yolo_object_detector.py:25]: Can we get rid of FasterRCNN maybe in a separate PR?

**Review by jarulraj — CHANGES_REQUESTED** [review]

**Comment by jarulraj** [comment]
The test cases are not being executed -- the formatting error has to be fixed first: https://app.circleci.com/pipelines/github/georgia-tech-db/eva/1946/workflows/751a8b36-3ae8-4996-9b20-e9b278571379/jobs/2752?invite=true#step-106-19

File: setup.py (hunk 4 above)
Comment (Aryan-Rajoria) [reply setup.py:101]: No Sir, `tensorboard` is a utility to monitor model performance on training and testing.
It is build to work without `tensorflow` library.

File: test/udfs/test_yolo_object_detector.py (hunk 5 above)
Comment (Aryan-Rajoria) [reply test/udfs/test_yolo_object_detector.py:25]: Yes Sir, we can do that.

File: eva/udfs/yolo_object_detector.py [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
+        Returns:
+            tuple containing predicted_classes (List[List[str]]),
+            predicted_boxes (List[List[BoundingBox]]),
+            predicted_scores (List[List[float]])
+
+        """
+        # Stacking all frames, and changing to numpy
+        # because of yolov5 error with Tensors
+
+        for i in range(frames.shape[0]):
```
Comment (gaurav274) [inline eva/udfs/yolo_object_detector.py:176]: Why aren't we doing batched inference? https://github.com/ultralytics/yolov5/issues/1806

**Review by gaurav274 — CHANGES_REQUESTED** [review]

File: eva/udfs/yolo_object_detector.py (hunk 6 above)
Comment (Aryan-Rajoria) [reply eva/udfs/yolo_object_detector.py:176]: Yolov5 requires the H and W of image to be a multiple of 32, however it is not guaranteed for all images. This causes `RuntimeError`.
However, when given images in `numpy`/`cv2 image` format it auto scales them to the proper format.
It also does not accept batch input in `numpy` format

**Review by jarulraj — APPROVED** [review]

**Pull request merged**